cmd/scollector/collectors/redis_unix.go

```diff
@@ -182,13 +180,22 @@ func c_redis() (opentsdb.MultiDataPoint, error) {
 			continue
 		}
 		defer c.Close()
-		lines, err := c.Do("INFO")
-		if err != nil {
-			Error = err
+
+		info_command, info_command_err := c.Do("info", "commandstats")
+		if info_command_err != nil {
+			Error = info_command_err
+			continue
+		}
+		info, info_err := c.Do("info")
+		if info_err != nil {
+			Error = info_err
 			continue
 		}
+
 		tags := instance.Copy()
-		infoSplit := strings.Split(string(lines.([]uint8)), "\n")
+		combined_lines := string(info.([]uint8)) + string(info_command.([]uint8))
```

Comment:
Why combine at all? Why not iterate over info_command separately? It seems like a lot of the conditional logic in later stages would be simpler by not combining.